### PR TITLE
Uf 5

### PR DESCRIPTION
**Pull request by mattrobertsky**

### CODE REVIEW

File: app/uk/gov/hmrc/digitalservicestax/controllers/ReturnsController.scala [hunk 1]
```diff
@@ -17,66 +17,75 @@
...
-import data.BackendAndFrontendJson._
-import config.AppConfig
-import connectors.{DSTConnector, MongoPersistence}
 import cats.implicits.catsKernelOrderingForOrder
-
-import javax.inject.Inject
-import ltbs.uniform._
-import common.web._
-import interpreters.playframework._
+import ltbs.uniform.{UniformMessages, _}
```
Comment (hmrclt) [inline app/uk/gov/hmrc/digitalservicestax/controllers/ReturnsController.scala:21]: import ltbs.uniform._

File: app/uk/gov/hmrc/digitalservicestax/controllers/ReturnsController.scala [hunk 2]
```diff
@@ -17,66 +17,75 @@
...
-      views.html.uniform.listing(rows.map {
-        case ListingTellRow(value, editLink, deleteLink) => (HtmlFormat.escape(value.name), editLink, deleteLink)
-      }, messages)
-  }
+  // private implicit def autoGroupListingTell = new ListingTell[Html, GroupCompany] {
+  //   def apply(rows: List[ListingTellRow[GroupCompany]], messages: UniformMessages[Html]): Html =
+  //     views.html.uniform.listing(rows.map {
+  //       case ListingTellRow(value, editLink, deleteLink) => (HtmlFormat.escape(value.name), editLink, deleteLink)
+  //     }, messages)
+  // }
```
Comment (hmrclt) [inline app/uk/gov/hmrc/digitalservicestax/controllers/ReturnsController.scala:77]: maybe remove this if it's not needed anymore?

File: app/uk/gov/hmrc/digitalservicestax/controllers/ReturnsController.scala [hunk 3]
```diff
@@ -90,14 +99,17 @@ class ReturnsController @Inject()(
 
   def showAmendments(): Action[AnyContent] = authorisedAction.async {
     implicit request: AuthorisedRequest[AnyContent] =>
-      implicit val msg: UniformMessages[Html] = interpreter.messages(request)
+      implicit val msg: UniformMessages[Html] = messages(request)
+
+      //TODO Check print feature from returns
+      //TODO Cache radio button selection
```
Comment (hmrclt) [inline app/uk/gov/hmrc/digitalservicestax/controllers/ReturnsController.scala:105]: are these important?

File: app/uk/gov/hmrc/digitalservicestax/journeys/RegJourney.scala [hunk 4]
```diff
@@ -26,47 +26,41 @@ import cats.implicits._
...
+  def registrationJourney[F[_] : Monad : TagK](
     backendService: DSTService[F]
-  ): F[Registration] = {
-    import interpreter._
+  ) = {
 
     for {
       globalRevenues <- ask[Boolean]("global-revenues")
-      _ <- if (!globalRevenues) { end("global-revenues-not-eligible", Kickout("global-revenues-not-eligible")) } else { (()).pure[F] }
+      _ <- end("global-revenues-not-eligible", Kickout("global-revenues-not-eligible")) unless globalRevenues
```
Comment (hmrclt) [inline app/uk/gov/hmrc/digitalservicestax/journeys/RegJourney.scala:49]: I think you should be able to do this in a single 'step' - 

```
_ <- end("global-revenues-not-eligible", Kickout("global-revenues-not-eligible")) unless ask[Boolean]("global-revenues")
```

you may prefer to keep it as two steps anyway, there's no performance advantage either way. 

you may also want to consider doing away with the Kickout class altogether - 

```
_ <- end("global-revenues-not-eligible") unless ask[Boolean]("global-revenues")
```

File: app/uk/gov/hmrc/digitalservicestax/journeys/RegJourney.scala [hunk 5]
```diff
@@ -26,47 +26,41 @@ import cats.implicits._
...
+      companyRegWrapper <- convert(backendService.lookupCompany()) flatMap { // gets a CompanyRegWrapper but converts to a company
 
         // found a matching company
         case Some(companyRW) =>
           for {
-            confirmCompany <- interact[Company, Boolean]("confirm-company-details", companyRW.company)
+            confirmCompany <- interact[Boolean]("confirm-company-details", companyRW.company)
             //TODO Here we need to sign the user out
-            _ <- if (!confirmCompany) { end("details-not-correct", Kickout("details-not-correct")) } else { (()).pure[F] }
+            _ <- if (!confirmCompany) { end("details-not-correct", Kickout("details-not-correct")) } else { pure(()) }
```
Comment (hmrclt) [inline app/uk/gov/hmrc/digitalservicestax/journeys/RegJourney.scala:59]:
```
_ <- end("details-not-correct", Kickout("details-not-correct")) unless confirmCompany
```

or even 

```
_ <- end("details-not-correct", Kickout("details-not-correct")) unless 
          interact[Boolean]("confirm-company-details", companyRW.company)
```

File: app/uk/gov/hmrc/digitalservicestax/journeys/RegJourney.scala [hunk 6]
```diff
@@ -167,15 +161,15 @@ object RegJourney {
...
+            pure(Option.empty[Company])
            }
           contactDetails <- ask[ContactDetails]("contact-details")
           groupMessage = if(isGroup) "group" else "company"
           liabilityDate <- ask[Boolean] (
             "check-liability-date",
             customContent = message("check-liability-date.required", groupMessage)
           ) flatMap {
-              case true => LocalDate.of(2020, 4, 1).pure[F]
+              case true => pure(LocalDate.of(2020, 4, 1))
```
Comment (hmrclt) [inline app/uk/gov/hmrc/digitalservicestax/journeys/RegJourney.scala:172]: is it fine to hard code a year in here?

File: app/uk/gov/hmrc/digitalservicestax/views/AdaptMessages.scala [hunk 7]
```diff
@@ -29,12 +29,14 @@ object AdaptMessages {
...
-      }.getOrElse(request.acceptLanguages.headOption.getOrElse(defaultLang))
-    }
+    def lang: Lang = defaultLang
+// TODO Use MessagesApi rather than Play
+//    {
+//      Play.maybeApplication.map { implicit app =>
+//        val maybeLangFromCookie = request.cookies.get(Play.langCookieName).flatMap(c => Lang.get(c.value))
+//        maybeLangFromCookie.getOrElse(defaultLang)
+//      }.getOrElse(request.acceptLanguages.headOption.getOrElse(defaultLang))
+//    }
```
Comment (hmrclt) [inline app/uk/gov/hmrc/digitalservicestax/views/AdaptMessages.scala:39]: something needs doing here?

File: app/uk/gov/hmrc/digitalservicestax/views/AdaptMessages.scala [hunk 8]
```diff
@@ -17,10 +17,10 @@
 package uk.gov.hmrc.digitalservicestax.views
 
 import java.util.Locale
-
 import ltbs.uniform._
 import play.api.Play
-import play.api.i18n.Messages.Implicits._
+import play.api.i18n.MessagesApi
+//import play.api.i18n.Messages.Implicits._
```
Comment (hmrclt) [inline app/uk/gov/hmrc/digitalservicestax/views/AdaptMessages.scala:23]: can this be removed?

File: app/uk/gov/hmrc/digitalservicestax/views/cya/CheckYourAnswersRet.scala.html [hunk 9]
```diff
@@ -0,0 +1,366 @@
...
+                    )
+                ))
+            )
+        ) ++ bankDetailsRows
+    }
+))
+@if(!isPrint) {
+    <h2 class="govuk-heading-m">@messages(s"$key.submission.title")</h2>
+    <p class="govuk-body">@messages(s"$key.submission.notice")</p>
+}
```
Comment (hmrclt) [inline app/uk/gov/hmrc/digitalservicestax/views/cya/CheckYourAnswersRet.scala.html:366]: can this be simplified a bit? It looks like there is a fair bit of repetition.

File: app/uk/gov/hmrc/digitalservicestax/views/uniform/Date.scala.html [hunk 10]
```diff
@@ -34,12 +37,13 @@
...
     @for((k,errorMessages) <- errors) {
         @for(errorMessage <- errorMessages.toList) {
-            @error_message(key.mkString(""), errorMessage.prefixWith(key ++ k.head).render(messages), messages)
+            @errorMessage.prefixWith(key ++ k.head).render(messages)
         }
     }
 }
 
 @if(key.length > 1) {
+    <h1>FOOBAR</h1>
```
Comment (hmrclt) [inline app/uk/gov/hmrc/digitalservicestax/views/uniform/Date.scala.html:46]: erm...

File: app/uk/gov/hmrc/digitalservicestax/views/uniform/FieldSurround.scala.html [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+        @*}*@
+
+        @**@
+        @** Will need to query if the content is a compound or standard field*@
+        @** so that we know how to wrap?*@
+        @**@
+        @*@content*@
+    @*} else { *@
+    @if(isCompoundField){
+@*        <div class="form-group@if(hasErrors){ form-group-error}">*@
```
Comment (hmrclt) [inline app/uk/gov/hmrc/digitalservicestax/views/uniform/FieldSurround.scala.html:72]: maybe delete this code if it's not needed

File: app/uk/gov/hmrc/digitalservicestax/views/uniform/FieldSurround.scala.html [hunk 12]
```diff
@@ -0,0 +1,146 @@
...
+@*                        <ul class="list list-bullet">*@
+@*                            @for(msg <- messages.list(s"$key.list")) {*@
+@*                                <li>@msg</li>*@
+@*                            }*@
+@*                        </ul>*@
+@*                    }*@
+
+@*                    @content*@
+@*                </fieldset>*@
+@*            </div>*@
```
Comment (hmrclt) [inline app/uk/gov/hmrc/digitalservicestax/views/uniform/FieldSurround.scala.html:126]: and this

File: app/uk/gov/hmrc/digitalservicestax/views/uniform/Listing.scala.html [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+@*                        index*@
+@*                    }">*@
+@*                        <span aria-hidden="true">@messages("common.delete")</span>*@
+@*                        <span class="visually-hidden">@messages("common.remove") @item</span>*@
+@*                    </a>*@
+@*                </dd>*@
+@*            </div>*@
+@*        }*@
+@*        </dl>*@
+@*    </div>*@
```
Comment (hmrclt) [inline app/uk/gov/hmrc/digitalservicestax/views/uniform/Listing.scala.html:58]: delete if not needed anymore

File: app/uk/gov/hmrc/digitalservicestax/views/uniform/StandardField.scala.html [hunk 14]
```diff
@@ -14,24 +14,27 @@
...
+@*//@noShowErrors=@{key == "manage-companies"}*@
+
+@*@if(hasFieldset) {*@
+@*    <div class="form-group@if(errors.definedAtRoot){ form-group-error}">*@
+@*        <fieldset*@
+@*        class="fieldset"*@
+@*        id="@{keyList.mkString("_")}-input"*@
+@*        @if(errors.definedAtRoot){aria-describedby='@errorKey'}*@
+@*        >*@
+@*}*@
```
Comment (hmrclt) [inline app/uk/gov/hmrc/digitalservicestax/views/uniform/StandardField.scala.html:37]: candidate for deletion

File: app/uk/gov/hmrc/digitalservicestax/views/uniform/StandardField.scala.html [hunk 15]
```diff
@@ -114,7 +87,7 @@ <h1 class="govuk-label-wrapper">
...
 
-@if(hasFieldset) {
-    </fieldset>
-</div>
-}
\ No newline at end of file
+@*@if(hasFieldset) {*@
+@*    </fieldset>*@
+@*</div>*@
+@*}*@
```
Comment (hmrclt) [inline app/uk/gov/hmrc/digitalservicestax/views/uniform/StandardField.scala.html:93]: delete if not needed

File: app/uk/gov/hmrc/digitalservicestax/views/uniform/StringField.scala.html [hunk 16]
```diff
@@ -0,0 +1,134 @@
...
+@telInput=@{key.contains("phoneNumber")}
+
+@autoComplete=@{key match {
+    case k if k.contains("forename") => "given-name"
+    case k if k.contains("surname") => "family-name"
+    case k if k.contains("phoneNumber") => "tel"
+    case k if k.contains("line1") => "address-line1"
+    case k if k.contains("line2") => "address-line2"
+    case k if k.contains("line3") => "address-line3"
+    case k if k.contains("line4") => "address-level4"
```
Comment (hmrclt) [inline app/uk/gov/hmrc/digitalservicestax/views/uniform/StringField.scala.html:53]: should that be line4?

File: app/uk/gov/hmrc/digitalservicestax/views/uniform/StringField.scala.html [hunk 17]
```diff
@@ -0,0 +1,134 @@
...
+@*                class="@{inputClass}@if(errors){ form-control-error}"*@
+@*                id="@{key.mkString("_")}-input"*@
+@*                @if(telInput){ inputmode="numeric" pattern="[0-9]*" }*@
+@*                name="@{key.mkString(".")}"*@
+@*                value="@{values}"*@
+@*                @autoFields.map { f => autocomplete=@f }*@
+@*                @if(autoComplete){autocomplete="@{autoComplete}"}*@
+@*                @if(currencyInput){inputMode="decimal"}*@
+@*                @if(percentageInput){inputMode="numeric" pattern="[0-9]*"}*@
+@*            />*@
```
Comment (hmrclt) [inline app/uk/gov/hmrc/digitalservicestax/views/uniform/StringField.scala.html:132]: delete if not needed

**Review by hmrclt — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/digitalservicestax/views/uniform/StringField.scala.html (hunk 16 above)
Comment (mattrobertsky) [reply app/uk/gov/hmrc/digitalservicestax/views/uniform/StringField.scala.html:53]: I'll double check this but the "address-level4" string is used elsewhere so assuming correct for now.

File: app/uk/gov/hmrc/digitalservicestax/journeys/RegJourney.scala [hunk 18]
```diff
@@ -26,47 +26,42 @@ import cats.implicits._
...
+  ) = {
 
     for {
-      globalRevenues <- ask[Boolean]("global-revenues")
-      _ <- if (!globalRevenues) { end("global-revenues-not-eligible", Kickout("global-revenues-not-eligible")) } else { (()).pure[F] }
-      ukRevenues <- ask[Boolean]("uk-revenues")
-      _ <- if (!ukRevenues) { end("uk-revenues-not-eligible", Kickout("uk-revenues-not-eligible")) } else { (()).pure[F] }
-
-      companyRegWrapper <- backendService.lookupCompany() >>= { // gets a CompanyRegWrapper but converts to a company
+//      globalRevenues <- ask[Boolean]("global-revenues")
```
Comment (hmrclt) [inline app/uk/gov/hmrc/digitalservicestax/journeys/RegJourney.scala:48]: you can take the commented bits out now if this works

File: app/uk/gov/hmrc/digitalservicestax/journeys/RegJourney.scala [hunk 19]
```diff
@@ -26,47 +26,42 @@ import cats.implicits._
...
+      companyRegWrapper <- convert(backendService.lookupCompany()) flatMap { // gets a CompanyRegWrapper but converts to a company
 
         // found a matching company
         case Some(companyRW) =>
           for {
-            confirmCompany <- interact[Company, Boolean]("confirm-company-details", companyRW.company)
+//            confirmCompany <- interact[Boolean]("confirm-company-details", companyRW.company)
             //TODO Here we need to sign the user out
-            _ <- if (!confirmCompany) { end("details-not-correct", Kickout("details-not-correct")) } else { (()).pure[F] }
+//            _ <- if (!confirmCompany) { end("details-not-correct", Kickout("details-not-correct")) } else { pure(()) }
```
Comment (hmrclt) [inline app/uk/gov/hmrc/digitalservicestax/journeys/RegJourney.scala:59]: another TODO, and commented lines

File: app/uk/gov/hmrc/digitalservicestax/views/uniform/Checkboxes.scala.html [hunk 20]
```diff
@@ -0,0 +1,85 @@
...
+      @*</label>*@
+    @*</div>*@
+
+    @*@if(conditional.isDefinedAt(opt)) {*@
+      @*<div id="@keyNoDots-@opt" class="checkboxes__conditional panel panel-border-narrow js-hidden">*@
+        @*@conditional(opt)*@
+      @*</div>*@
+    @*}*@
+  @*}*@
+  @*</div>*@
```
Comment (hmrclt) [inline app/uk/gov/hmrc/digitalservicestax/views/uniform/Checkboxes.scala.html:84]: another candidate for deletion

File: project/metals.sbt [hunk 21]
```diff
@@ -0,0 +1,4 @@
+// DO NOT EDIT! This file is auto-generated.
+// This file enables sbt-bloop to create bloop config files.
+
+addSbtPlugin("ch.epfl.scala" % "sbt-bloop" % "1.4.8-19-4d9f966b")
```
Comment (hmrclt) [inline project/metals.sbt:4]: I'd put metals.sbt under .gitignore

**Review by hmrclt — CHANGES_REQUESTED** [review]

**Review by hmrclt — APPROVED** [review]

**Pull request merged**